types/experiments.ts

```diff
@@ -58,7 +58,10 @@ export interface NimbusExperiment {
   featureIds?: Array<string>;
 
   /** Branch configuration for the experiment */
-  branches: Array<Branch>;
+  branches:
+    | Array<SingleFeatureBranch>
+    | Array<MultiFeatureDesktopBranch>
+    | Array<MultiFeatureMobileBranch>;
```

Comment:
Nits: This looks weird. Not necessarily wrong, just weird.

```ts
branches: Array<Branch>
…
type Branch = SingleFeatureBranch | MultiFeatureBranch | DeprecatedMultiFeatureBranch
```

i.e. 

 * an Array<T>, where T = U,V,W;
 * would expect the Deprecated branch type to be more easily removable.

Reply:
Yeah agreed, or even just `Array<SingleFeatureBranch | MultiFeatureBranch | DeprecatedMultiFeatureBranch>`

I like the use of the word deprecated.

Reply:
@k88hudson Putting the type union inside the array would I think mean you could have a different types in the same array, I put the arrays in the union so all branches would have to be the same type in the array.